packages/components/src/MessageBar/MessageBar.tsx

```diff
@@ -239,6 +240,24 @@ const MessageBarContent = styled.div`
   padding: 0 ${({ theme: { space } }) => space.large};
 `
 
+const backgroundColor = variant({
+  prop: 'intent',
+  variants: {
+    critical: {
+      backgroundColor: 'criticalAccent',
+    },
+    inform: {
+      backgroundColor: 'informAccent',
+    },
+    positive: {
+      backgroundColor: 'positiveAccent',
+    },
+    warn: {
+      backgroundColor: 'warnAccent',
+    },
+  },
+})
+
 export const MessageBar = styled(MessageBarLayout).attrs(
   ({ px = 'medium', py = 'small', width = '100%' }) => ({
     px,
```

Comment:
You can use Styled Components `attrs` method to inject default `intent` value:

```suggestion
  ({ intent = 'inform', px = 'medium', py = 'small', width = '100%' }) => ({
    intent,
    px,
```